### ISSUE TITLE
Multiple `references()` in the same table, only the first is actually created

### ISSUE THREAD

**Issue by hmeranda** [simolus3/moor#2292]
Worked in flutter 3.3.9, drift 2.1
Not working on Flutter 3.7, drift 2.5

warning on code generation about field2 and  field3: "The referenced element, OtherTable, is not understood by drift."
(no warning on field1)
logging the create table also confirms, only the first FK is created (for field1).

for example (simplified):
```
class OtherTable extends Table {
  TextColumn get id => text()();

  @override
  Set<Column> get primaryKey => {id};
}

class ThisTable extends Table {
  TextColumn get field1 => text().references(OtherTable, #id)();
  TextColumn get field2 => text().references(OtherTable, #id)();
  TextColumn get field3 => text().references(OtherTable, #id)();
}
```

help appreciated, thank you.

**Comment by simolus3** [comment]
Thanks for the report! Unfortunately, I couldn't reproduce the issue. With this file, there were no warnings, the generated code looked correct and the created tables all referenced the right column:

```dart
import 'package:drift/drift.dart';
import 'package:drift/native.dart';

part 'repro.g.dart';

class OtherTable extends Table {
  TextColumn get id => text()();

  @override
  Set<Column> get primaryKey => {id};
}

class ThisTable extends Table {
  TextColumn get field1 => text().references(OtherTable, #id)();
  TextColumn get field2 => text().references(OtherTable, #id)();
  TextColumn get field3 => text().references(OtherTable, #id)();
}

@DriftDatabase(tables: [OtherTable, ThisTable])
class Database extends _$Database {
  Database(super.e);

  @override
  int get schemaVersion => 1;
}

void main() async {
  final db = Database(NativeDatabase.memory(logStatements: true));
  await db.thisTable.all().get();
}
```

**Comment by North101** [comment]
Usually when this kind of issue pops up, it's because custom constraints are also being used which replace any other constraint. This isn't in your example, but perhaps that's what's happening in your code?

**Comment by hmeranda** [comment]
@simolus3 thank you very much for the quick response,
i'll try and isolate this into an example project that reproduces the issue and post it here.

@North101 i also have some constraints (indexes) defined in a `drift` file, maybe it causes this.

**Comment by hmeranda** [comment]
@simolus3 @North101  i've managed to reproduce this by splitting the code into multiple files, as well as creating some indexes via a drift file.
not sure which addition causes this, and maybe i'm doing something incorrectly, just tried to make it similar to my original code.

thanks again for your help


main.dart
```dart
import 'package:drift/drift.dart';
import 'package:drift/native.dart';

import 'other_table.dart';
import 'this_table.dart';

part 'main.g.dart';

@DriftDatabase(tables: [OtherTable, ThisTable], include: {"extras.drift"})
class Database extends _$Database {
  Database(super.e);

  @override
  int get schemaVersion => 1;
}

void main() async {
  final db = Database(NativeDatabase.memory(logStatements: true));
  await db.thisTable.all().get();
}
```

this_table.dart
```dart
import 'package:drift/drift.dart';

import 'other_table.dart';

class ThisTable extends Table {
  TextColumn get id => text()();

  TextColumn get field1 => text().references(OtherTable, #id)();
  TextColumn get field2 => text().references(OtherTable, #id)();
  TextColumn get field3 => text().references(OtherTable, #id)();

  TextColumn get field4 => text()();

  @override
  Set<Column> get primaryKey => {id};
}
```

other_table.dart
```dart
import 'package:drift/drift.dart';

class OtherTable extends Table {
  TextColumn get id => text()();

  TextColumn get field5 => text()();

  @override
  Set<Column> get primaryKey => {id};
}
```

extras.drift
```
import 'this_table.dart';
import 'other_table.dart';

CREATE INDEX this_table_index ON this_table (field4);

CREATE INDEX other_table_index ON other_table (field5);
```

code generation warnings (notice there are no warnings on field1):
```
[INFO] 10.5s elapsed, 0/5 actions completed.
[WARNING] drift_dev on lib/this_table.dart:
line 10, column 46 of package:tem/this_table.dart: The referenced element, OtherTable, is not understood by drift.
   ╷
10 │   TextColumn get field2 => text().references(OtherTable, #id)();
   │                                              ^^^^^^^^^^
   ╵
[WARNING] drift_dev on lib/this_table.dart:
line 11, column 46 of package:tem/this_table.dart: The referenced element, OtherTable, is not understood by drift.
   ╷
11 │   TextColumn get field3 => text().references(OtherTable, #id)();
   │                                              ^^^^^^^^^^
   ╵
[INFO] Running build completed, took 10.9s

```

execution log (notice only field1 got an FK)
```
I/flutter (17330): Drift: Sent CREATE TABLE IF NOT EXISTS "other_table" ("id" TEXT NOT NULL, "field5" TEXT NOT NULL, PRIMARY KEY ("id")); with args []
I/flutter (17330): Drift: Sent CREATE TABLE IF NOT EXISTS "this_table" ("id" TEXT NOT NULL, "field1" TEXT NOT NULL REFERENCES other_table (id), "field2" TEXT NOT NULL, "field3" TEXT NOT NULL, "field4" TEXT NOT NULL, PRIMARY KEY ("id")); with args []
I/flutter (17330): Drift: Sent CREATE INDEX this_table_index ON this_table (field4) with args []
I/flutter (17330): Drift: Sent CREATE INDEX other_table_index ON other_table (field5) with args []
```

**Comment by simolus3** [comment]
Thanks, I was able to reproduce that one. Should be fixed in `drift_dev`  version `2.5.1`.

**Comment by hmeranda** [comment]
> Thanks, I was able to reproduce that one. Should be fixed in `drift_dev`  version `2.5.1`.

Much appreciated
